### ISSUE TITLE
GUI does not build on Fedora 25

### ISSUE THREAD

**Issue by susilehtola** [openbabel/openbabel#1574]
Steps to reproduce:
1. Check out repository
2. mkdir objdir && cd objdir
3. cmake -DCMAKE_INSTALL_PREFIX=$(pwd)/../install ..
4. make -j9 install VERBOSE=1

Expected result:
openbabel builds and installs.

Actual result:
Build fails with
```
[ 86%] Linking CXX executable ../../bin/obgui
cd /home/work/openbabel/objdir/src/GUI && /usr/bin/cmake -E cmake_link_script CMakeFiles/obgui.dir/link.txt --verbose=1
/usr/lib64/ccache/c++    -pthread -O2 -DNDEBUG -O3 -DNDEBUG  -Wl,--enable-new-dtags -Wl,--fatal-warnings -Wl,--no-undefined -lc -Wl,--enable-new-dtags -Wl,--fatal-warnings -Wl,--no-undefined -lc  CMakeFiles/obgui.dir/OBGUI.o CMakeFiles/obgui.dir/stdwx.o CMakeFiles/obgui.dir/optswx.o  -o ../../bin/obgui -rdynamic ../../lib/libopenbabel.so.5.0.0 -pthread -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -lwx_baseu-2.8 -lwx_gtk2u_core-2.8 -lwx_gtk2u_adv-2.8 -lm -ldl -lz -Wl,-rpath,/home/work/openbabel/objdir/lib: 
/usr/bin/ld: CMakeFiles/obgui.dir/OBGUI.o: relocation R_X86_64_32 against `_ZN16SelFormatsDialog13sm_eventTableE' can not be used when making a shared object; recompile with -fPIC
CMakeFiles/obgui.dir/OBGUI.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
src/GUI/CMakeFiles/obgui.dir/build.make:149: recipe for target 'bin/obgui' failed
make[2]: *** [bin/obgui] Error 1
```

edit: it does work if -DBUILD_SHARED=OFF (in this case GUI is not built).

**Comment by baoilleach** [comment]
Which OS?

**Comment by susilehtola** [comment]
Fedora 25, and yes.

**Comment by baoilleach** [comment]
Does it fail elsewhere if you turn off building the GUI?

**Comment by susilehtola** [comment]
If the GUI is turned off, the build succeeds.

**Comment by baoilleach** [comment]
Ok - now I'm confused. Did the Fedora folk already manage to compile this (https://apps.fedoraproject.org/packages/openbabel-gui) and if so, how did they do that?

**Comment by susilehtola** [comment]
I have no idea why the build has worked before; to my knowledge you *always* need a -fPIE or -fPIC to make a valid shared library...

**Comment by baoilleach** [comment]
Fixed by #1575. Thanks.
